### DISCUSSION TITLE
Red Knot Contributions

### DISCUSSION THREAD

**Discussion by Slyces** [astral-sh/ruff#12143] [General]
Hi guys, first off, congratulation on the awesome work you've been doing.
I have seen that ruff is currently working on type checking capabilities with red-knot.
Do you guys welcome contributions to this project, even on very minor issues?
I'd love to do what I can to make this happen sooner 😄

**Comment by MichaReiser** [comment]
Hi @Slyces and awesome that you're interested in contributing. 

Contributing is a bit hard at the moment because we have yet to figure out the fundamental architecture fully. That means there's a lot of change in direction, unstable APIs, merge conflicts. Coordination is a challenge right now. That's why we aren't currently looking for external contributors (we haven't onboarded the entire team for the very same reason). 

I would be very interested in external contributions once we reach a step where things feel "stable enough" and where we also have work that isn't on the critical path, where having it a day or two later immediately becomes a blocker for the team. Give us another 4 weeks and we hopefully have most figured (there will still be ton of work!) out by then.

**Reply by Slyces** [reply]
Noted, thank you!

**Reply by purepani** [reply]
Just wanted to quickly ask if there have been any updates on red knot! There seem to be a lot of PRs lately!

**Reply by MichaReiser** [reply]
Thanks for the ping! I wanted to post an update but forgot. I'm sorry. 

You can use [this view](https://github.com/astral-sh/ruff/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+label%3Ared-knot) to see all red-knot issues. There's at least one that's currently marked as help wanted and there are a few bugs that might not be too hard. We plan to create more good first issues/help wanted issues in the near time.

**Reply by ewilazarus** [reply]
Hey, I'd like to ask if this is already usable? Are there any docs?

(Sorry if this is not the right thread)

**Reply by purepani** [reply]
Afaik, no it is not.
Its still very early in development

**Reply by winwinashwin** [reply]
Hi, I would like to do some early experiments with red-knot-wasm. Is red-knot currently in a usable state to test things out?

**Reply by MichaReiser** [reply]
No, not yet. There will be an official announcement when it is, but I suspect that it will take even longer before the WASM api becomes stable (it's mainly for internal use)
